File: LeetCode/66. Plus One.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
    var inc = 1;
    digits = digits.reverse().map(num => {
        if (num + inc >= 10) {
            let tmp = (num + inc) % 10;
            inc = Math.floor((num + inc) / 10);
            return tmp;
        } else {
            let tmp = num + inc;
            inc = Math.floor((num + inc) / 10);
            return tmp;
        }
    });

    if (inc) {
        digits.push(inc);
    }

    return digits.reverse();
};

File: LeetCode/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 1;

        reverse(digits.begin(), digits.end());
        for (int i = 0; i < digits.size(); i++) {
            carry += digits[i];
            if (carry >= 10) {
                digits[i] = carry % 10;
                carry /= 10;
            } else {
                digits[i] = carry;
                carry = 0;
            }
        }
        if (carry) {
            digits.push_back(carry);
        }
        reverse(digits.begin(), digits.end());
        return digits;
    }
};
